src/main/java/au/org/emii/ncdfgenerator/NcdfEncoderBuilder.java

```diff
@@ -0,0 +1,1476 @@
+
+
+package au.org.emii.ncdfgenerator;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream ;
+import java.io.OutputStream ;
+import java.io.IOException;
+
+import java.nio.charset.Charset;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern ;
+import java.util.regex.Matcher;
+import java.util.AbstractMap.SimpleImmutableEntry;
+
+import java.sql.*;
+
+import java.text.SimpleDateFormat;
+
+import ucar.nc2.NetcdfFileWriteable;
+import ucar.nc2.Dimension;
+import ucar.ma2.DataType;
+import ucar.ma2.Array;
+import ucar.ma2.ArrayDouble;
+import ucar.ma2.Index;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.NodeList;
+
+import java.nio.charset.StandardCharsets;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+
+
+class NcdfGeneratorException extends Exception
+{
+    public NcdfGeneratorException(String message)
+	{
+        super(message);
+    }
+}
+
+
+
+interface IExprVisitor
+{
+	public void visit( ExprInteger expr )throws Exception;
+	public void visit( ExprProc expr ) throws Exception;
+	public void visit( ExprLiteral expr ) throws Exception;
+	public void visit( ExprSymbol expr ) throws Exception;
+	public void visit( ExprTimestamp expr )throws Exception;
+}
+
+
+interface IExpression
+{
+	public int getPosition() ;
+	public void accept( IExprVisitor v ) throws Exception ;
+}
+
+
+class ExprSymbol implements IExpression
+{
+	public ExprSymbol( int pos, String value)
+	{
+		this.pos = pos;
+		this.value = value;
+	}
+
+	public int getPosition() { return pos; }
+	public void accept( IExprVisitor v )  throws Exception
+	{ v.visit( this); }
+
+	final int pos;
+	final String value;
+}
+
+
+class ExprInteger implements IExpression
+{
+	public ExprInteger( int pos, int value)
+	{
```

Comment:
Perhaps readability could be improved here by modifying variable names here for each class? Sorry @julian1 if it's on your list of things to do!